packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -742,8 +769,8 @@ mixin WidgetInspectorService {
   /// The VM service protocol does not keep alive object references so this
   /// class needs to manually manage groups of objects that should be kept
   /// alive.
-  final Map<String, Set<_InspectorReferenceData>> _groups = <String, Set<_InspectorReferenceData>>{};
-  final Map<String, _InspectorReferenceData> _idToReferenceData = <String, _InspectorReferenceData>{};
+  final Map<String, Set<InspectorReferenceData>> _groups = <String, Set<InspectorReferenceData>>{};
+  final Map<String, InspectorReferenceData> _idToReferenceData = <String, InspectorReferenceData>{};
   final Map<Object, String> _objectToId = Map<Object, String>.identity();
```

Comment:
this needs to change as otherwise you are holding onto all of the objects defeating the point of using weakref.
Idea: use a Map for only the objects that cannot work with expando and place and use Expando for the other objects. Alternately you could use Finalizer to create a Weakmap but that would be less efficient.